main/provider/index.ts

```diff
@@ -928,15 +921,16 @@ export class Provider extends EventEmitter {
   private parseTargetChain (payload: RPCRequestPayload) {
     const target: Chain = { type: 'ethereum', id: 0 }
 
-    if (!('chainId' in payload)) {
-      return { ...target, id: store('main.currentNetwork.id') }
-    }
-
-    const chainId = parseInt(payload.chainId || '', 16)
-    const chainConnection = this.connection.connections['ethereum'][chainId] || {}
+    if ('chainId' in payload) {
+      const chainId = parseInt(payload.chainId || '', 16)
+      const chainConnection = this.connection.connections['ethereum'][chainId] || {}
 
-    if (chainConnection.chainConfig) {
-      target.id = chainId
+      if (chainConnection.chainConfig) {
+        target.id = chainId
+      }
+    } else {
+      const chainId = store('main.dapps', payload._origin, 'chainId')
+      target.id = chainId || 1
```

Comment:
I don't think we should default to mainnet here, as this should never really be undefined and if it is something is wrong and we don't want to route requests to mainnet by accident. I think returning an error is probably the way to go

Reply:
Down to 5 broken now

Reply:
The error handling is already covered by line 954 so I just removed the default